### PR TITLE
Task 6

### PR DESCRIPTION
**Pull request by qhussain97**

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -89,6 +88,42 @@ describe('GET /api/articles/:article_id', () => {
...
-
-
+describe('GET /api.articles/:article_id/comments', () => {
+    test('status 200: responds with an array of comments for given article_id', () => {
+        return request(app)
+        .get('/api/articles/3/comments')
+        .expect(200)
+        .then((response) => {
+            const comments = response.body.comments;
+            console.log(response.body)
```
Comment (lewispricey) [inline __tests__/app.test.js]: Please remove the console.log

File: __tests__/app.test.js [hunk 2]
```diff
@@ -89,6 +88,42 @@ describe('GET /api/articles/:article_id', () => {
...
+            comments.forEach((comment) => {
```
Comment (lewispricey) [inline __tests__/app.test.js:100]: It would be great to check the length of the array is expected for this one to prevent any false positives before your loop

File: controllers/controllers.ncnews.js [hunk 3]
```diff
@@ -22,4 +23,15 @@ exports.getArticleById = (req, res, next) => {
         .catch((err) => {
             next(err);
         });
+}
+
+exports.getCommentsForArticleId = (req, res, next) => {
+    const article_id = req.params.article_id;
+    selectCommentsForArticleId(article_id)
+        .then((comments) => {
+            res.status(200).send({ comments })
```
Comment (lewispricey) [inline controllers/controllers.ncnews.js]: Good descriptive key to send the data back on!

File: models/models.js [hunk 4]
```diff
@@ -30,4 +31,20 @@ exports.selectArticleById = (article_id) => {
...
+
+exports.selectCommentsForArticleId = (article_id) => {
+    let queryString = `
+        SELECT comments.comment_id, comments.body, comments.votes, comments.created_at, comments.author
+        FROM comments
+        WHERE article_id = $1
+        ORDER BY comments.created_at DESC`;
+    return db.query(queryString, [article_id])
+        .then((results) => {
+            if (results.rowCount === 0) {
```
Comment (lewispricey) [inline models/models.js]: Would this cause problems if the article exists but doesn't have any comments?

File: __tests__/app.test.js [hunk 5]
```diff
@@ -89,6 +88,42 @@ describe('GET /api/articles/:article_id', () => {
         });
     })
 })
-});
-
-
+describe('GET /api.articles/:article_id/comments', () => {
```
Comment (lewispricey) [inline __tests__/app.test.js:92]: We could do with an extra test for a 200 - empty array if the article exists but doesn't have any comments

**Review by lewispricey — CHANGES_REQUESTED** [review]
Great work so far Qasim!

Just need to add in a test and some logic to differenciate between the comments being empty because there isn't any comments for the article and the article not existing in the database. 

You might want to think back to Jim's lecture on Monday where he used the promise.all with the help of another of the functions he wrote.

**Review by Dutton1990 — APPROVED** [review]
Thanks for making those changes Qasim, looks great to me. Nice job! You can merge and move on 😄
